gossip/src/crds_value.rs

```diff
@@ -204,6 +209,47 @@ impl SnapshotHash {
         }
     }
 }
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, AbiExample)]
+pub struct IncrementalSnapshotHash {
```

Comment:
I am also wondering what exactly are we saving here by not including the base slot hash here, ie.
```rust
base: (Slot, Hash),
```

Reply:
Yeah, this was something I went back and forth on a few times.

When creating the actual incremental snapshot archive (which is where the snapshot hashes are pushed to CRDS), the incremental snapshot package does not know the full snapshot's hash, just the slot. So, if we want to have the base Hash here too, then either (1) the base (full) snapshot hash needs to be looked up* when pushing to CRDS, or (2) the incremental snapshot package needs to have the full snapshot hash added to it (which would need changes in AccountsBackgroundService to keep that info and put it into the incremental snapshot package).

Both of those options are possible, but the full snapshot hash isn't needed here, since we also have the `SnapshotHashes` for full snapshots.

*To handle the look up, this is somewhat tricky. We could look up the full snapshot archives on disk and then find the one with the matching slot and get its hash. I had an impl a while back that did this for a different reason, but design feedback was to avoid going to disk to do the lookup, for correctness and performance.

Since bootstrap doesn't need `IncrementalSnapshotHashes` to have the base hash, and the Incremental Snapshot in concept doesn't need the base hash, I didn't add it. If later there's a new unified snapshot hashes type for both full and incremental snapshots, then I think we'd get the base slot and hash for free.

I understand that putting the base hash here could be used during bootstrap, but I don't know of a reason that the base hash _must_ be here. Is there a use-case that you have in mind that would need the base hash here?

Reply:
I think the benefit would be that it will make the struct self-contained; and does not introduce cross-dependency between crds values.
It shouldn't be too much work for whatever the code which receives base-slot to also do the book-keeping for the snapshot hash for that slot. no?

Reply:
Done! I've updated the struct to also include the hash for the base as well.

Reply:
We're good with only supporting one full snapshot that incrementals can be based on?  I can't think of anything immediately problematic with this.  Just pointing out that we're about to design away that possibility.

Reply:
Yeah, creating the incremental snapshot requires specifying what (single) slot it is based on. So the impl would have to change quite a lot to support something else.

Reply:
Oh, I think I misinterpreted the question. Yeah I see what you mean. I originally had _every_ IncrementalSnapshotHash also store its full snapshot (i.e. base) slot. We could do that, but it just means we consume more bandwidth.

Reply:
That, or it could be a HashMap of base slot to everything. I.e.

```struct IncrementalSnapshotHahes {
    from: Pubkey,
    hashes: HashMap<Slot, (Hash, (Slot, Hash)),
    wallclock: u64,
}
```